src/main/java/net/fabricmc/loader/EntrypointStorage.java

```diff
@@ -203,4 +205,23 @@ boolean hasEntrypoints(String key) {
 
 		return results;
 	}
+
+	protected <T> List<LazyEntrypointContainer<T>> getLazyEntrypointContainers(String key, Class<T> type) {
```

Comment:
A behavior change where existing entrypoints will not create the instance until you ask for it via `getEntrypoint()` is what I'd prefer probably. You can't really do much via relying on the ctor of the entrypoint in order to instantiate everything.

In the case of patchwork, you'd obviously get what you want with a behavior change. If you are relying on the instance being created when you resolve the list then I feel that is a horrible hack.

Reply:
I agree, the time of construction is an impl detail. I can make this change tomorrow.